package.json

```diff
@@ -56,49 +56,49 @@
-    "@jupyterlab/codeeditor": "3.5.0",
-    "@jupyterlab/docmanager": "3.5.0",
-    "@jupyterlab/docregistry": "3.5.0",
-    "@jupyterlab/filebrowser": "3.5.0",
-    "@jupyterlab/fileeditor": "3.5.0",
-    "@jupyterlab/launcher": "3.5.0",
-    "@jupyterlab/mainmenu": "3.5.0",
-    "@jupyterlab/markdownviewer": "3.5.0",
-    "@jupyterlab/notebook": "3.5.0",
-    "@jupyterlab/observables": "4.5.0",
-    "@jupyterlab/rendermime": "3.5.0",
-    "@jupyterlab/services": "6.5.0",
-    "@jupyterlab/settingregistry": "3.5.0",
-    "@jupyterlab/shared-models": "3.5.0",
-    "@jupyterlab/ui-components": "3.5.0",
-    "@lumino/coreutils": "1.12.1",
-    "@lumino/signaling": "1.11.0",
-    "@lumino/widgets": "1.35.0"
+    "@jupyterlab/application": "^3.5.0",
+    "@jupyterlab/apputils": "^3.5.0",
+    "@jupyterlab/codeeditor": "^3.5.0",
+    "@jupyterlab/docmanager": "^3.5.0",
+    "@jupyterlab/docregistry": "^3.5.0",
+    "@jupyterlab/filebrowser": "^3.5.0",
+    "@jupyterlab/fileeditor": "^3.5.0",
+    "@jupyterlab/launcher": "^3.5.0",
+    "@jupyterlab/mainmenu": "^3.5.0",
+    "@jupyterlab/markdownviewer": "^3.5.0",
+    "@jupyterlab/notebook": "^3.5.0",
+    "@jupyterlab/observables": "^4.5.0",
+    "@jupyterlab/rendermime": "^3.5.0",
+    "@jupyterlab/services": "^6.5.0",
+    "@jupyterlab/settingregistry": "^3.5.0",
+    "@jupyterlab/shared-models": "^3.5.0",
+    "@jupyterlab/ui-components": "^3.5.0",
+    "@lumino/coreutils": "^1.12.1",
+    "@lumino/signaling": "^1.11.0",
+    "@lumino/widgets": "^1.35.0"
   },
   "devDependencies": {
-    "@babel/core": "7.20.2",
+    "@babel/core": "^7.20.2",
     "@babel/preset-env": "^7.20.2",
-    "@jupyterlab/builder": "3.5.0",
-    "@jupyterlab/testutils": "3.5.0",
-    "@types/jest": "26.0.24",
-    "@typescript-eslint/eslint-plugin": "4.33.0",
-    "@typescript-eslint/parser": "4.33.0",
-    "eslint": "7.32.0",
-    "eslint-config-prettier": "6.15.0",
-    "eslint-plugin-prettier": "3.4.1",
-    "jest": "26.6.3",
-    "npm-run-all": "4.1.5",
-    "prettier": "2.7.1",
-    "rimraf": "3.0.2",
-    "stylelint": "14.14.1",
-    "stylelint-config-prettier": "9.0.4",
-    "stylelint-config-recommended": "6.0.0",
-    "stylelint-config-standard": "24.0.0",
-    "stylelint-prettier": "2.0.0",
-    "ts-jest": "26.5.6",
-    "typescript": "4.1.6"
+    "@jupyterlab/builder": "^3.5.0",
+    "@jupyterlab/testutils": "^3.5.0",
+    "@types/jest": "^26.0.24",
+    "@typescript-eslint/eslint-plugin": "^4.33.0",
+    "@typescript-eslint/parser": "^4.33.0",
+    "eslint": "^7.32.0",
+    "eslint-config-prettier": "^6.15.0",
+    "eslint-plugin-prettier": "^3.4.1",
+    "jest": "^26.6.3",
+    "npm-run-all": "^4.1.5",
+    "prettier": "^2.7.1",
+    "rimraf": "^3.0.2",
+    "stylelint": "^14.14.1",
+    "stylelint-config-prettier": "^9.0.4",
+    "stylelint-config-recommended": "^6.0.0",
+    "stylelint-config-standard": "^24.0.0",
+    "stylelint-prettier": "^2.0.0",
+    "ts-jest": "^26.5.6",
+    "typescript": "~4.1.6"
```

Comment:
You should always specify a range version for dependencies.
`^` pins the major version but all the others can be upgraded
`~` pins the major and minor version